src/embind/embind_shared.js

```diff
@@ -145,7 +145,7 @@ var LibraryEmbindShared = {
 #if ASSERTIONS
       assert(signature[signature.length - 1] == ")", "Parentheses for argument names should match.");
 #endif
-      return signature.substr(argsIndex, signature.length - argsIndex - 1).replaceAll(" ", "").split(",").filter(n => n.length);
+      return signature.substr(argsIndex, signature.length - argsIndex - 1).match(/\w+(?=\s*(,|$))/g) || [];
```

Comment:
Can you explain what this regex is doing? Perhaps in a comment on the line above?

Do we need to worry about complex cases where the itself contains a comma (e.g. `std::pair<int, int>`)?   This kind of case seems extremely tricky.

For this reason I wonder if we should reconsider and just accept a list of names, without types?   (especially since we don't actually need the types themselves here do we?  (i.e. the types comes via a different mechanism, right?)

Reply:
If we drop the types then we can also drop the type checking. 

That only leaves the minor issue or not being able to copy and paste directly, which I don't think is such a large advantage to warrant this much churn and a potentially breaking change (since we might already have users of these argument names, no?)

Reply:
It's not a breaking change as omitting the types is still allowed when using emscripten::function directly. The existing test cases all still pass, unmodified. Also I did a search on GitHub and didn't find a single case of anyone using this feature, probably because it's undocumented.

The type checking is one of my main motivations for this change. I am maintaining bindings for a C++ library. When I update the bindings to a new version of the library I want compile errors when existing functions change, so that I can be sure to update the bindings. Otherwise they are sure to get out of date.

Reply:
To be clear the types themselves will automatically updated right?   The part that needs to be manually updated is that names for the arguments?     Would it be enough to perhaps just check the number of arguments?     That might be enough to catch a lot of common cases where the names need updating?

Reply:
> Also I did a search on GitHub and didn't find a single case of anyone using this feature, probably because it's undocumented.

I see.. but to be clear if anyone was using this feature then this change would break their build?  Is that right?

Reply:
> if anyone was using this feature then this change would break their build? Is that right?

No, that is incorrect. Again, the existing tests pass unmodified.

Reply:
> Would it be enough to perhaps just check the number of arguments? That might be enough to catch a lot of common cases where the names need updating?

Yes it will catch some cases. Type checking will catch more.

Obviously ideally this would check the actual argument names from the function's own implementation. Unfortunately this would require C++ reflection and it is not clear when or if that will ever be available.